.github/workflows/ci-llama.yaml

```diff
@@ -75,10 +75,16 @@ jobs:
             iree-runtime \
             "numpy<2.0"
 
-      - name: Run llama test
+      - name: Run llama 8b tests
+        if: github.event_name == "pull_request"
+        run: pytest sharktank/tests/models/llama/benchmark_amdgpu_test.py -v -s --iree-hip-target=gfx942
+
+      - name: Run llama 70b and 405b tests
+        if: github.event_name != "pull_request"
         run: pytest sharktank/tests/models/llama/benchmark_amdgpu_test.py -v -s --longrun --iree-hip-target=gfx942 --html=out/index.html
```

Comment:
Changing which tests a workflow runs based on the event trigger feels like a recipe for confusion.
* What happens with the `workflow_dispatch` trigger?
* Will developers know that https://github.com/nod-ai/SHARK-Platform/actions/workflows/ci-llama.yaml?query=event%3Apull_request and https://github.com/nod-ai/SHARK-Platform/actions/workflows/ci-llama.yaml?query=event%3Aschedule are testing entirely different things?

Workflow should be as simple and predictable as possible - don't overengineer this.

I would split into one workflow that runs on `push` and `pull_request` then a separate workflow that runs on `schedule` (and both should support `workflow_dispatch` for debugging).

If you're concerned about duplicating boilerplate... that needs to be fixed too. The nightly workflow should be using nightly release packages, and we need to roll out the new packaging into these workflows. Lines 59-76 should be condensed down to one line.

Reply:
alright sounds good, will split into two workflow files

Reply:
If the tests fit nicely alongside other sharktank tests, whatever runs on `push` and `pull_request` could be part of https://github.com/nod-ai/SHARK-Platform/blob/main/.github/workflows/ci-sharktank.yml.

What were the original reasons for creating a separate workflow?

Reply:
The original reason for creating a separate workflow was to be able to have a few of the small llama tests (8b) run on `pull_request` in order to test any regressions in sharktank with all the changes coming in. The larger llama tests (70b and 405b) take much longer, so would just be run nightly.

CC @saienduri

Reply:
Yeah let's go with two workflows due to the machine constraint (won't be able to slot nicely into ci-sharktank).

Reply:
Can you clarify "machine constraint"? We should have a matrix that runs over different configurations (CPU on standard runners, GPU on self-hosted MI300, etc.). Each test can use marks to specify what it requires, how long it takes, etc. and then the workflow can choose what subset of tests to run by passing those marks.

I don't want to end up with too many workflows. There should be an obvious answer to "is this test running on CI, and if so, which workflow?" for every test.

For example:
* `ci-sharktank.yml`: run all tests under sharktank/ on whatever hardware we have, except large tests and benchmarks
* `ci-sharktank-nightly.yml`: run large tests and benchmarks under sharktank/

Reply:
The main thing is the hardcoded paths, so it would have to be it's own step that only runs when the `matrix.runs_on` is the mi300x machine, which I guess is fine. For the long tests, we only want to run that on a schedule, so having that in its own workflow file makes the most sense.

Reply:
So, Avi we can probably add the 8b testing to the `ci-sharktank.yml` and the full llama testing to a new workflow file `ci-sharktank-nightly.yml`

Reply:
Well we shouldn't have any hardcoded paths. That needs to be fixed. Write tests _first_ for developers to use (on any system that has the right hardware, enough disk space, etc.), _then_ teach the CI to run them.

Reply:
@aviator19941 is it possible to have the 8b files in huggingface?

Reply:
Fully agree, but looks like there are too many moving parts/files/configurations right now (model size, data type, batch size, TP sharding, decomposed vs decodeposed attention) which makes it hard to move to huggingface quickly. Let's go with separate workflows for now (just so it doesn't halt testing coverage for llama), but the end goal here should be moving to huggingface and the ci-sharktank.yml.